### ISSUE TITLE
Crash to desktop upon program launch

### ISSUE THREAD

**Issue by Vezzra** [freeorion/freeorion#1197]
### Environment

* **FreeOrion Version**: f6038bb5ee896795ac9fb126637aa7ed9ad52001
* **Operating System**: Windows
* **Fetched as**
  * Compiled from source

### Description

When I try to launch FO on Windows, I get the following message:

![message](https://cloud.githubusercontent.com/assets/10279417/21643187/4487b230-d287-11e6-8835-c4f96811ee23.png)

Sorry for the german message, IIRC in English it's "FreeOrion.exe has stopped working".

The "problem details" are reported as follows:
```
Problemsignatur:
  Problemereignisname:	APPCRASH
  Anwendungsname:	FreeOrion.exe
  Anwendungsversion:	0.0.0.0
  Anwendungszeitstempel:	586cee80
  Fehlermodulname:	FreeOrion.exe
  Fehlermodulversion:	0.0.0.0
  Fehlermodulzeitstempel:	586cee80
  Ausnahmecode:	c0000005
  Ausnahmeoffset:	00445f01
  Betriebsystemversion:	6.1.7601.2.1.0.768.3
  Gebietsschema-ID:	3079
  Zusatzinformation 1:	0a9e
  Zusatzinformation 2:	0a9e372d3b4ad19135b953a78882e789
  Zusatzinformation 3:	0a9e
  Zusatzinformation 4:	0a9e372d3b4ad19135b953a78882e789
```
The commit that introduced the issue is f6038bb5ee896795ac9fb126637aa7ed9ad52001 (@adrianbroher, that's one of yours).

Call stack I got from the MSVC 2015 debugger:
```
>	FreeOrion.exe!??@574a78b846d43084cb18325c73d036db@(boost::xpressive::detail::match_state<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & state, const boost::xpressive::detail::stacked_xpression<boost::xpressive::detail::stacked_xpression<boost::xpressive::detail::static_xpression<boost::xpressive::detail::independent_end_matcher,boost::xpressive::detail::no_next>,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_matcher<boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::posix_charset_matcher<boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> >,boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::regex_matcher<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_end_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_end_matcher<boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > > > > >,boost::fusion::nil_> >,boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_end_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_end_matcher<boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > > > > >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_end_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_end_matcher<boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > > > & next) Line 104	C++
 	FreeOrion.exe!boost::xpressive::detail::repeat_begin_matcher::match<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > >,boost::xpressive::detail::stacked_xpression<boost::xpressive::detail::static_xpression<boost::xpressive::detail::independent_end_matcher,boost::xpressive::detail::no_next>,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_matcher<boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::posix_charset_matcher<boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> >,boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::regex_matcher<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_end_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_end_matcher<boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > > > > >,boost::fusion::nil_> >,boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_end_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_end_matcher<boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > > > > > >(boost::xpressive::detail::match_state<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & state, const boost::xpressive::detail::stacked_xpression<boost::xpressive::detail::static_xpression<boost::xpressive::detail::independent_end_matcher,boost::xpressive::detail::no_next>,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_matcher<boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::posix_charset_matcher<boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> >,boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_begin_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::regex_matcher<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_end_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_end_matcher<boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > > > > >,boost::fusion::nil_> >,boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::mark_end_matcher,boost::xpressive::detail::static_xpression<boost::xpressive::detail::repeat_end_matcher<boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > > > > > & next) Line 55	C++
 	FreeOrion.exe!??@7ef4d5b41461a76f533ae118e84ff381@(boost::xpressive::detail::match_state<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & state, const boost::xpressive::detail::static_xpression<boost::xpressive::detail::regex_matcher<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::simple_repeat_matcher<boost::xpressive::detail::static_xpression<boost::xpressive::detail::posix_charset_matcher<boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::true_matcher,boost::xpressive::detail::no_next> >,boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_matcher<boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::literal_matcher<boost::xpressive::cpp_regex_traits<char>,boost::mpl::bool_<0>,boost::mpl::bool_<0> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> >,boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::regex_matcher<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> >,boost::fusion::nil_> >,boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_matcher<boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::string_matcher<boost::xpressive::cpp_regex_traits<char>,boost::mpl::bool_<0> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::regex_matcher<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::string_matcher<boost::xpressive::cpp_regex_traits<char>,boost::mpl::bool_<0> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::simple_repeat_matcher<boost::xpressive::detail::static_xpression<boost::xpressive::detail::posix_charset_matcher<boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::true_matcher,boost::xpressive::detail::no_next> >,boost::mpl::bool_<1> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::literal_matcher<boost::xpressive::cpp_regex_traits<char>,boost::mpl::bool_<0>,boost::mpl::bool_<0> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > > > > >,boost::xpressive::detail::alternates_list<boost::xpressive::detail::static_xpression<boost::xpressive::detail::regex_matcher<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::literal_matcher<boost::xpressive::cpp_regex_traits<char>,boost::mpl::bool_<0>,boost::mpl::bool_<0> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::alternate_end_matcher,boost::xpressive::detail::no_next> > >,boost::fusion::nil_> >,boost::xpressive::cpp_regex_traits<char> >,boost::xpressive::detail::static_xpression<boost::xpressive::detail::end_matcher,boost::xpressive::detail::no_next> > > > > & next, boost::mpl::bool_<0> __formal) Line 72	C++
 	FreeOrion.exe!??@f1c8bdcb1b97e0c04e985845b767ff0a@(boost::xpressive::detail::match_state<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & state) Line 53	C++
 	FreeOrion.exe!boost::xpressive::basic_regex<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >::match_(boost::xpressive::detail::match_state<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & state) Line 233	C++
 	FreeOrion.exe!boost::xpressive::detail::regex_search_impl<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >(boost::xpressive::detail::match_state<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & state, const boost::xpressive::basic_regex<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & re, bool not_initial_null) Line 321	C++
 	FreeOrion.exe!boost::xpressive::regex_search<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > >(std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > begin, std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > end, boost::xpressive::match_results<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & what, const boost::xpressive::basic_regex<std::_String_const_iterator<std::_String_val<std::_Simple_types<char> > > > & re, boost::xpressive::regex_constants::match_flag_type flags) Line 442	C++
 	FreeOrion.exe!StringTable_::Load(const StringTable_ * lookups_fallback_table) Line 103	C++
 	FreeOrion.exe!StringTable_::StringTable_(const std::basic_string<char,std::char_traits<char>,std::allocator<char> > & filename, const StringTable_ * lookups_fallback_table) Line 27	C++
 	FreeOrion.exe!`anonymous namespace'::GetStringTable(std::basic_string<char,std::char_traits<char>,std::allocator<char> > stringtable_filename) Line 31	C++
 	FreeOrion.exe!UserString(const std::basic_string<char,std::char_traits<char>,std::allocator<char> > & str) Line 58	C++
 	FreeOrion.exe!MessageWnd::MessageWnd(const std::basic_string<char,std::char_traits<char>,std::allocator<char> > & config_name) Line 308	C++
 	FreeOrion.exe!ClientUI::ClientUI() Line 603	C++
 	FreeOrion.exe!HumanClientApp::HumanClientApp(int width, int height, bool calculate_fps, const std::basic_string<char,std::char_traits<char>,std::allocator<char> > & name, int x, int y, bool fullscreen, bool fake_mode_change) Line 235	C++
 	FreeOrion.exe!mainSetupAndRun() Line 264	C++
 	FreeOrion.exe!wmain(int argc, wchar_t * * argv, wchar_t * * envp) Line 101	C++
```

**Comment by Vezzra** [comment]
Updated OP with call stack.

**Comment by geoffthemedio** [comment]
Hopefully fixed with fdf1f1435fdd5be2974a91576ee7ffda13faf278 (by reverting part of it). Don't understand the issue, but don't want to wait for another solution.

**Comment by Vezzra** [comment]
Yep, the fix works. Thanks!
